File: web-client/src/lib/types/notification.ts
import { EntitySentiment } from './article'

export interface NotificationDto {
    id: number
    ticker: string
    article_url: string
    article_processed_at: string
    sentiment_score: number
    sentiment_label: string
    is_read: boolean
    created_at: string
    source?: string | null
    article_title?: string | null
    article_overall_sentiment_score?: number | null
    article_overall_sentiment_label?: string | null
    article_entities?: EntitySentiment[] | null
}

File: core-api/src/main/java/org/finsight/coreapi/domain/Notification.java
package org.finsight.coreapi.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.OffsetDateTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "notifications")
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String ticker;

    @Column(name = "article_url", nullable = false)
    private String articleUrl;

    @Column(name = "article_processed_at", nullable = false)
    private OffsetDateTime articleProcessedAt;

    @Column(nullable = false)
    private Double sentimentScore;

    @Column(nullable = false)
    private String sentimentLabel;

    @Builder.Default
    private Boolean isRead = false;

    @Builder.Default
    private OffsetDateTime createdAt = OffsetDateTime.now();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumns({
            @JoinColumn(name = "article_url", referencedColumnName = "url", insertable = false, updatable = false),
            @JoinColumn(name = "article_processed_at", referencedColumnName = "processed_at", insertable = false, updatable = false)
    })
    private Article article;
}
